src/datachain/lib/dc.py

```diff
@@ -1286,6 +1287,52 @@ def show(
         if len(df) == limit:
             print(f"\n[Limited by {len(df)} rows]")
 
+    @classmethod
+    def from_hf(
+        cls,
+        dataset: Union[str, "HFDatasetType"],
+        session: Optional[Session] = None,
+        object_name: str = "",
+        model_name: str = "",
+        **kwargs,
+    ) -> "DataChain":
+        """Generate chain from huggingface hub dataset.
+
+        Parameters:
+            dataset : Path or name of the dataset to read from Hugging Face Hub,
+                or an instance of `datasets.Dataset`-like object.
+            session : Session to use for the chain.
+            object_name : Generated object column name.
+            model_name : Generated model name.
+            kwargs : Parameters to pass to datasets.load_dataset.
+
+        Example:
+            Load from Hugging Face Hub:
+            ```py
+            DataChain.from_hf("beans", split="train")
+            ```
+
+            Generate chain from loaded dataset:
+            ```py
+            from datasets import load_dataset
+            ds = load_dataset("beans", split="train")
+            DataChain.from_hf(ds)
+            ```
+        """
+        from datachain.lib.hf import HFGenerator, get_output_schema, stream_dataset
+
+        if isinstance(dataset, str):
+            dataset = stream_dataset(dataset, **kwargs)
+
+        model_name = model_name or object_name or ""
+        output = get_output_schema(dataset, model_name)
+        model = dict_to_data_model(model_name, output)
+        if object_name:
+            output = {object_name: model}
+
+        chain = DataChain.from_records(DataChain.DEFAULT_FILE_RECORD, session=session)
+        return chain.gen(HFGenerator(dataset, model), output=output)
```

Comment:
[F] From working with the hf examples there is a trick that you can use to download from the hub much quicker. From https://github.com/iterative/datachain/pull/264: "Using the `hf_transfer` library and `HF_HUB_ENABLE_HF_TRANSFER` environment variable within that example means it will run within a reasonable time."

Reply:
As long as `HF_HUB_ENABLE_HF_TRANSFER` is set before you import any of the hf packages you should see much quicker transfer speeds.

Reply:
We are streaming the dataset here. Do you know if it makes a difference in that case? Hopefully it's less of a problem. We could require `huggingface_hub[hf_transfer]` but not sure it's necessary. Feels like this should be up to HF to make the default.

Reply:
https://github.com/huggingface/huggingface_hub/issues/2279